lib/ExpensiMark.d.ts

```diff
@@ -0,0 +1,118 @@
+declare type Replacement = string | ((...args: string[]) => string);
```

Comment:
I saw that this type is being used in `rules` array and other methods like `modifyTextForUrlLinks`, `modifyTextForEmailLinks`, `modifyTextForQuote` and `formatTextForQuote`. The `rules` array indeed accept both string and function, but the other methods only uses `replacement` as a function. We should have a separate type for them.

Reply:
I agree, adjusted the code 🙆‍♂️